fabric-block-api-v1/src/main/java/net/fabricmc/fabric/mixin/block/ChunkSectionBlockStateCounterMixin.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.block;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+
+@Mixin(targets = "net/minecraft/world/chunk/ChunkSection$BlockStateCounter")
+public class ChunkSectionBlockStateCounterMixin {
+	/**
+	 * Makes Chunk Sections not have isAir = true modded blocks be replaced with AIR against their will.
+	 * Mojang report: https://bugs.mojang.com/browse/MC-232360
+	 */
+	@Redirect(method = "accept(Lnet/minecraft/block/BlockState;I)V",
+			at = @At(value = "INVOKE", target = "Lnet/minecraft/block/BlockState;isAir()Z"))
+	private boolean modifyAirCheck(BlockState blockState) {
+		return blockState.isOf(Blocks.AIR) || blockState.isOf(Blocks.CAVE_AIR) || blockState.isOf(Blocks.VOID_AIR);
```

Comment:
Question: Is there a concern about peformance here? I havent looked at the taregt code, but I expect they are quite hot. There are prob a few ways this could be made faster, at least to match the previous .isAir() call?

Reply:
The isOf call is just a reference check so I don't think it should be much concern. 
![image](https://github.com/FabricMC/fabric/assets/40846040/51c4a0ad-e6e4-4e7d-9738-cb415e2f573d)

air will match first check and exit. Non-air would go through all 3. I could add a `isAir() && (the three vanilla air checks) ` instead so non-air exits on first check and AIR exits on second check. And cave_air, void_air and modded air has to pass all 4 checks

Reply:
Doesn't Lithium touch this code?

Reply:
Not the isAir checks
https://github.com/CaffeineMC/lithium-fabric/blob/b8f2818899bfcdae1bbf34991f2d3351e72c2f6c/src/main/java/me/jellysquid/mods/lithium/mixin/chunk/no_locking/ChunkSectionMixin.java#L20

https://github.com/CaffeineMC/lithium-fabric/blob/b8f2818899bfcdae1bbf34991f2d3351e72c2f6c/src/main/java/me/jellysquid/mods/lithium/mixin/util/block_tracking/ChunkSectionMixin.java#L28

Reply:
After some discussion in discord, the three isOf checks should be extremely fast. The call a method that calls a final field so isOf should be considered a pure future by JVM and condensed into a single call or something. Optimization is weird. Even if it didn't, trying to micro-optimize it might only shave a few nanoseconds which won't matter much here.

@modmuss50 If this sounds good, i can mark this convo as resolved

Reply:
Sure, just wanted to talk to someone who knew more about this than I do. Can always change it later if it does become a worry.